UPGRADE_GUIDE.md

```diff
@@ -2,22 +2,130 @@
 
 Use the following guide to assist in the upgrade process of the `easypost-java` library between major versions.
 
-* [Upgrading from 4.x to 5.0](#upgrading-from-4x-to-50)
+- [Upgrading from 5.x to 6.0](#upgrading-from-5x-to-60)
+- [Upgrading from 4.x to 5.0](#upgrading-from-4x-to-50)
+
+## Upgrading from 5.x to 6.0
+
+### 6.0 High Impact Changes
+
+- [Client Instance](#60-client-instance)
+- [Error Types](#60-error-types)
+- [Uniform empty response](#60-uniform-empty-response)
+
+### 6.0 Low Impact Changes
+
+- [Services and Resources](#60-services-and-resources)
+
+## 6.0 Client Instance
+
+*Likelihood of Impact: **High***
+
+The library is now designed around the idea of a `EasyPostClient`. Users will initialize a `EasyPostClient` instance with their API key
+and then use that instance to make API calls.
+
+This change allows for multiple clients to be instantiated with different API keys and for the library to be used in a
+multi-threaded environment.
+
+```java
+// Old
+EasyPost.apiKey = System.getenv("EASYPOST_API_KEY"); // global API key
+
+// New
+EasyPostClient client1 = new EasyPostClient("EASYPOST_API_KEY_1"); // per-client API key
+EasyPostClient client2 = new EasyPostClient("EASYPOST_API_KEY_2"); // per-client API key
+```
+
+All functions are now accessed via the `EasyPostClient` instance. For example, to create a shipment:
+
+```java
+// Old
+EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");
+Shipment shipment = Shipment.create(shipmentMap);
+
+// New
+EasyPostClient client = new EasyPostClient("EASYPOST_API_KEY_1");
+Shipment shipment = client.shipment.create(shipmentMap);
+```
+
+## 6.0 Error Types
+
+*Likelihood of Impact: **High***
+
+Error handling has been overhauled and a number of specific exception types have been introduced.
+
+All exceptions inherit from `EasyPost.Exception.EasyPostException` (which extends `System.Exception`). Users can catch this exception type to handle all errors thrown by the library.
+
+Subclasses of `EasyPostException` are grouped into two categories:
```

Comment:
- Was all of this copied from the C# guide? 
- If so, are we sure we included every error type it had? 
- Did we double check that the syntax is correct for Java?

Reply:
Yes, it was copied from the C# guide but I have doubled check that every error type is included. I will remove the `EasyPost.Exception` namespace from the nested list since they are not necessarily needed.